### PR TITLE
Rename lifetime member to Pocket Casts Champion 

### PR DESCRIPTION
**Pull request by mebarbosa**
## Description
- Rename `Lifetime member` to `Pocket Casts Champion`
- Opens a dialog when the user taps on `Pocket Casts Champion`
- See PT: pdeCcb-67m-p2
- See figma: G60UqxPhTynbalo0Vfr1NC-fi-2515_9059

## Testing Instructions
1. Log with a lifetime user account. See PdNHOb-2Q-p2
2. Go to profile and tap on `Pocket Casts Champion`
3. ✅ Ensure you see the dialog
4. Tap on Rate button
5. ✅ Ensure you are redirected to play store

## Screenshots or Screencast 

image (1)

## Checklist
- [x] If this is a user-facing change, I have added an entry in CHANGELOG.md
- [x] Ensure the linter passes (`./gradlew spotlessApply` to automatically apply formatting/linting)
- [x] I have considered whether it makes sense to add tests for my changes
- [x] All strings that need to be localized are in `modules/services/localization/src/main/res/values/strings.xml`
- [x] Any jetpack compose components I added or changed are covered by compose previews
- [x] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.
 
#### I have tested any UI changes...

- [x] with different themes
- [x] with a landscape orientation
- [x] with the device set to have a large display and font size
- [x] for accessibility with TalkBack

### CODE REVIEW

File: modules/services/analytics/src/main/java/au/com/shiftyjelly/pocketcasts/analytics/TracksAnalyticsTracker.kt [hunk 1]
```diff
@@ -81,7 +81,7 @@ class TracksAnalyticsTracker @Inject constructor(
             PredefinedEventProperty.HAS_DYNAMIC_FONT_SIZE to displayUtil.hasDynamicFontSize(),
             PredefinedEventProperty.USER_IS_LOGGED_IN to isLoggedIn,
             PredefinedEventProperty.PLUS_HAS_SUBSCRIPTION to hasSubscription,
-            PredefinedEventProperty.PLUS_HAS_LIFETIME to hasLifetime,
+            PredefinedEventProperty.PLUS_HAS_LIFETIME to isPocketCastsChampion,
```
Comment (mebarbosa) [inline modules/services/analytics/src/main/java/au/com/shiftyjelly/pocketcasts/analytics/TracksAnalyticsTracker.kt:84]: We agreed that we are not going to update this PLUS_HAS_LIFETIME event name. See p1721917283521179/1720493861.494549-slack-C05RR9P9RAT

File: modules/features/profile/src/main/java/au/com/shiftyjelly/pocketcasts/profile/champion/PocketCastsChampionBottomSheetDialog.kt [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        return ComposeView(context).apply {
+            setContent {
+                AppTheme(theme.activeTheme) {
+                    CallOnce {
+                        analyticsTracker.track(AnalyticsEvent.POCKET_CASTS_CHAMPION_DIALOG_SHOWN)
+                    }
+                    ChampionDialog(
+                        onRateClick = {
+                            analyticsTracker.track(AnalyticsEvent.POCKET_CASTS_CHAMPION_DIALOG_RATE_BUTTON_TAPPED)
+                            rateUs(context)
```
Comment (ashiagr) [inline modules/features/profile/src/main/java/au/com/shiftyjelly/pocketcasts/profile/champion/PocketCastsChampionBottomSheetDialog.kt:40]: Does it make sense to show an in-app rating dialog here?
If we agree to show it, changes need to be made to allow the dialog to be shown even if it was shown from the Stats screen earlier.

Comment (mebarbosa) [reply modules/features/profile/src/main/java/au/com/shiftyjelly/pocketcasts/profile/champion/PocketCastsChampionBottomSheetDialog.kt:40]: Great question. I tried to make it simpler, but I raised a question about it here pdeCcb-67m-p2#comment-4847

**Review by ashiagr — APPROVED** [review]

File: modules/features/profile/src/main/java/au/com/shiftyjelly/pocketcasts/profile/champion/PocketCastsChampionBottomSheetDialog.kt (hunk 2 above)
Comment (mebarbosa) [reply modules/features/profile/src/main/java/au/com/shiftyjelly/pocketcasts/profile/champion/PocketCastsChampionBottomSheetDialog.kt:40]: I've just confirmed that we are going to keep this as it is pdeCcb-67m-p2#comment-4850

**Comment by MtndaleRedneck** [comment]
It's not bad enough that you've been screwing over all your early supporters that bought the app but didn't buy the web by giving them no acknowledgment and making them pay for the app. Now you go and and change the name from Lifetime user to Champion and add language about how only those users have been with you since the beginning. What about all the users who bought the app?

**Comment by yaelirub** [comment]
We care deeply about all users and work hard to give everyone the features they love. We're constantly working to make the user experience better free or premium. This is just a change in terminology to give these users a hat tip. Nothing else.

**Comment by MtndaleRedneck** [comment]
But what about all the users who paid for the app? Why don't they get any sort of hat tip? Why were they just kicked to the curb when it was inconvenient?

**Comment by yaelirub** [comment]
As mentioned, we're dedicated to providing features that all our users love. Changing the terminology for one group of people does not affect another, nor does it imply any broader changes.

**Comment by krisdb** [comment]
Imagine your life is so boring you get upset about not getting a "hat tip".

**Pull request merged**